src/test/java/com/polishuchenko/bookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,247 @@
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CategoryControllerTest {
+    private static MockMvc mockMvc;
+    private static final String ADD_CATEGORY_SCRIPT =
+            "classpath:database/categories/add-category-detective-to-category-table.sql";
+    private static final String DELETE_CATEGORY_SCRIPT =
+            "classpath:database/categories/remove-category-detective-from-category-table.sql";
+    private static final String ADD_BOOK_SCRIPT =
+            "classpath:database/books/add-book-kobzar-to-book-table.sql";
+    private static final String DELETE_BOOK_SCRIPT =
+            "classpath:database/books/delete-book-kobzar-from-book-table.sql";
+    private static final String ADD_CATEGORY_BOOK_SCRIPT =
+            "classpath:database/books_categories/add-kobzar-detective-to-category_book-table.sql";
+    private static final String DELETE_CATEGORY_BOOK_SCRIPT =
+            "classpath:database/books_categories/delete-from-category_book-table.sql";
+    private static final String USER_NAME = "user";
+    private static final String ADMIN_NAME = "admin";
+    private static final Long VALID_ID = 1L;
+    private static final Long INVALID_ID = 99L;
+    private static CategoryDto detectiveResponse;
+    private static CategoryRequestDto validRequestDto;
+    private static CategoryRequestDto invalidRequestDto;
+    private static BookDtoWithoutCategoryIds kobzar;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+
+        kobzar = new BookDtoWithoutCategoryIds();
+        kobzar.setId(1L);
+        kobzar.setTitle("Kobzar");
+        kobzar.setAuthor("Taras Shevchenko");
+        kobzar.setIsbn("978-3-16-148410-0");
+        kobzar.setPrice(BigDecimal.valueOf(50));
+        kobzar.setDescription("some description");
+        kobzar.setCoverImage("some link");
+
+        detectiveResponse = new CategoryDto();
+        detectiveResponse.setId(VALID_ID);
+        detectiveResponse.setName("detective");
+        detectiveResponse.setDescription("some description");
+
+        validRequestDto = new CategoryRequestDto();
+        validRequestDto.setName(detectiveResponse.getName());
+        validRequestDto.setDescription(detectiveResponse.getDescription());
+
+        invalidRequestDto = new CategoryRequestDto();
+        invalidRequestDto.setName("");
+    }
+
+    @WithMockUser(username = ADMIN_NAME, roles = {"ADMIN"})
+    @Test
+    @Sql(
+            scripts = DELETE_CATEGORY_SCRIPT, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @DisplayName("Create new category and add to DB")
+    void create_validRequest_ReturnsCategoryDto() throws Exception {
```

Comment:
```suggestion
    void create_validRequest_returnsCategoryDto() throws Exception {
```
after `_` use lowercase. fix it in all places